File: utility/templates/solution_template.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
)

func readInput() []string {
	f, err := os.Open("input")
	if err != nil {
		log.Fatalf("open input: %v", err)
	}
	defer f.Close()

	var lines []string
	sc := bufio.NewScanner(f)
	for sc.Scan() {
		lines = append(lines, sc.Text())
	}
	if err := sc.Err(); err != nil {
		log.Fatalf("scan input: %v", err)
	}
	return lines
}

func partOne(lines []string) string {
	// TODO: implement
	return fmt.Sprintf("part 1 not implemented (%d lines)", len(lines))
}

func partTwo(lines []string) string {
	// TODO: implement
	return fmt.Sprintf("part 2 not implemented (%d lines)", len(lines))
}

func main() {
	lines := readInput()
	if len(os.Args) > 1 {
		if os.Args[1] == "1" {
			fmt.Println(partOne(lines))
			return
		}
		if os.Args[1] == "2" {
			fmt.Println(partTwo(lines))
			return
		}
	}
	fmt.Println(partOne(lines))
	fmt.Println(partTwo(lines))
}

File: utility/templates/solution_template.rs
use std::env;
use std::fs;

fn read_input() -> Vec<String> {
    let data = fs::read_to_string("input").expect("failed to read input");
    data.lines().map(|s| s.to_string()).collect()
}

fn part_one(lines: &[String]) -> String {
    // TODO: implement
    format!("part 1 not implemented ({} lines)", lines.len())
}

fn part_two(lines: &[String]) -> String {
    // TODO: implement
    format!("part 2 not implemented ({} lines)", lines.len())
}

fn main() {
    let lines = read_input();
    let arg = env::args().nth(1);
    match arg.as_deref() {
        Some("1") => println!("{}", part_one(&lines)),
        Some("2") => println!("{}", part_two(&lines)),
        _ => {
            println!("{}", part_one(&lines));
            println!("{}", part_two(&lines));
        }
    }
}
